src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/handlers/presubmit/letter/HomeOfficeNonStandardDirectionLetterGenerator.java

```diff
@@ -0,0 +1,86 @@
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.field.Document;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.handlers.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.service.DocumentCreator;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.service.DocumentHandler;
+import uk.gov.hmcts.reform.iacasedocumentsapi.infrastructure.DirectionFinder;
+
+@Component
+public class HomeOfficeNonStandardDirectionLetterGenerator implements PreSubmitCallbackHandler<AsylumCase> {
+
+    private final DocumentCreator<AsylumCase> homeOfficeNonStandardDirectionLetterCreator;
+    private final DocumentHandler documentHandler;
+    private final DirectionFinder directionFinder;
+
+    public HomeOfficeNonStandardDirectionLetterGenerator(
+        @Qualifier("homeOfficeNonStandardDirectionLetter")
+        DocumentCreator<AsylumCase> homeOfficeNonStandardDirectionLetterCreator,
+        DocumentHandler documentHandler,
+        DirectionFinder directionFinder) {
+        this.homeOfficeNonStandardDirectionLetterCreator = homeOfficeNonStandardDirectionLetterCreator;
+        this.documentHandler = documentHandler;
+        this.directionFinder = directionFinder;
+    }
+
+    public boolean canHandle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+            && callback.getEvent() == Event.SEND_DIRECTION
+            && isInternalCase(asylumCase)
+            && isAppellantInDetention(asylumCase)
+            && isRecipientRespondent(asylumCase);
+    }
+
+    public PreSubmitCallbackResponse<AsylumCase> handle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+        final CaseDetails<AsylumCase> caseDetails = callback.getCaseDetails();
+        final AsylumCase asylumCase = caseDetails.getCaseData();
+
+        Document uploadDocument = homeOfficeNonStandardDirectionLetterCreator.create(caseDetails);
+        documentHandler.addWithMetadata(
+            asylumCase,
+            uploadDocument,
+            TRIBUNAL_DOCUMENTS,
+            DocumentTag.IA_HOME_OFFICE_NON_STANDARD_DIRECTION_LETTER
+        );
+
+        return new PreSubmitCallbackResponse<>(asylumCase);
+    }
+
+    private boolean isRecipientRespondent(AsylumCase asylumCase) {
```

Comment:
ive seen this repeated in a few PRs. Maybe we can move it out to the utility class to avoid code duplications.